distribution/bwc/build.gradle

```diff
@@ -29,14 +29,33 @@ import java.util.regex.Matcher
  * tests to test against the next unreleased version, closest to this version,
  * without relying on snapshots.
  */
-final Matcher match = project.name =~ /bwc-snapshot-(\d+\.(\d+|x))/
-if (!match.matches()) {
-  throw new InvalidUserDataException("Unsupport project name ${project.name}")
-}
-String bwcBranch = match.group(1)
+subprojects {
+
+  Version bwcVersion
+  if (project.name == 'next-minor-snapshot') {
+    bwcVersion = versionCollection.nextMinorSnapshot
+  } else if (project.name == 'staged-minor-snapshot') {
+    bwcVersion = versionCollection.stagedMinorSnapshot
+  } else if (project.name == 'maintenance-bugfix-snapshot') {
+    bwcVersion = versionCollection.maintenanceBugfixSnapshot
+  } else if (project.name == 'next-bugfix-snapshot') {
+    bwcVersion = versionCollection.nextBugfixSnapshot
```

Comment:
As above, I think it'd be nice to move the logic that relates Gradle project names and the versions all the way into `VersionCollection`.

Reply:
yea i think its a good idea to leave it as a helper method in the VC code